benchmarks/nemo/nemolite2d/manual_versions/psykal_kokkos/Makefile

```diff
@@ -10,23 +10,56 @@
 # export OMPFLAGS=" -fopenmp"
 # export AR=ar
 
-# Additionally it needs the path to a Kokkos installation and execution
-# devices
-KOKKOS_PATH ?= ${HOME}/build/kokkos
-KOKKOS_DEVICES=OpenMP  # Currently this can only be OpenMP
+# Additionally it needs the path to the Kokkos source code
+KOKKOS_PATH ?= $(HOME)/tools/kokkos
+KOKKOS_DEBUG ?= yes
+
+# If no KOKKOS_DEVICES is specified, by default use the OpenMP
+ifndef $(KOKKOS_DEVICES)
+KOKKOS_DEVICES=OpenMP
+endif
+
+# Configuration specific to the selected Kokkos device 
+ifeq ($(KOKKOS_DEVICES),OpenMP)
+$(info "Using OpenMP device")
+# KOKKOS_OPTIONS := "aggresive_vectorization"
+else ifeq ($(KOKKOS_DEVICES),Serial)
+$(info "Using Serial device")
+# KOKKOS_OPTIONS := "aggresive_vectorization"
+else ifeq ($(KOKKOS_DEVICES),Pthreads)
+$(error "Pthreads device is currently not supported")
+else ifeq ($(KOKKOS_DEVICES),Cuda)
+$(info "Using CUDA device")
+KOKKOS_ARCH = Pascal61
+# nvcc doesn't support all compiler flags, overwirte CFLAGS to just O3
+CFLAGS := "-O3"
+# Still use the selected compiler but using the Kokkos nvcc_wrapper
+NVCC_WRAPPER_DEFAULT_COMPILER = $(CXX)
+CXX := $(KOKKOS_PATH)/bin/nvcc_wrapper
+KOKKOS_CUDA_OPTIONS = "enable_lambda"
+else
+$(error "Unrecognised KOKKOS_DEVICES value: $(KOKKOS_DEVICES)")
+endif
 
 .PHONY: nemolite2d_kokkos
-.DEFAULT_GOAL := nemolite2d_kokkos
+.DEFAULT_GOAL := nemolite2d_views_kokkos
```

Comment:
This target needs to be added to the .PHONY list and `nemolite2d_kokkos` removed.

Reply:
oops I forgot about the PHONY list, there is in fact a few targets that are missing. Fixed